snapcraft/plugins/dotnet.py

```diff
@@ -140,3 +151,64 @@ def _get_appname(self):
             if fnmatch.fnmatch(file, '*.??proj'):
                 return os.path.splitext(file)[0]
                 break
+
+    def _get_version_metadata(self, version):
+        jsonData = self._get_dotnet_release_metadata()
+        package_data = list(filter(lambda x: x.get('version-runtime')
+                            == version, jsonData))
+
+        if not package_data or len(package_data) < 1:
+            print('error occurred while fetching  the version details'
+                  'or the version specified is incorrect')
```

Comment:
This will just carry on its merry way-- should an error be raised instead to halt things? An error that preferably inherits from `SnapcraftError`. Otherwise we'll just get an out-of-bounds error [here](https://github.com/snapcore/snapcraft/pull/1911/files#diff-5d24e70cc05a6003012e19fc045d6592R179).

Reply:
Indeed. I didn't want to blow up this PR but I went ahead and added classes for the errors.